scripts/make_confusion_matrix.py

```diff
@@ -0,0 +1,89 @@
+import argparse
+import json
+from pathlib import Path
+
+import numpy as np
+import sklearn
+import sklearn.metrics
+
+
+allowed_metrics = ["acc", "acc_norm"]
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--file",
+        required=True,
+        metavar="DIR|DIR/file.json",
+        help="Path of json file containing sample predictions.",
+    )
+    parser.add_argument(
+        "--metric",
+        default="acc",
+        choices=allowed_metrics,
+        help="Method of determining prediction.",
+    )
+    return parser.parse_args()
+
+
+def confusion_matrix(file: str, metric: str):
+    file = Path(file)
+    assert file.exists(), f"File does not exist! ({file})"
+
+    sample_dump = []
+    with open(file, "r") as f:
+        for line in f:
+            sample = json.loads(line)
+            sample_dump.append(sample)
+
+    choices = [d["arg_1"].strip() for d in sample_dump[0]["arguments"].values()]
+    completion_len = np.array([float(len(i)) for i in choices])
+    print(f"Classes: {choices}")
+
+    targets, predictions = [], []
+    for sample in sample_dump:
+        target = sample["target"]
+        if isinstance(target, int):
+            target_label = choices[target]
+            targets.append(target_label)
+        elif isinstance(target, str):
+            assert target in choices
+            targets.append(target)
+        elif isinstance(target, list):
+            raise NotImplementedError(
+                "No support yet for multi-label confusion matrix!"
+            )
```

Comment:
Target can be specified as`int` (index of true label) or `str` (true label) or `list` (multi-label tasks). I decided to not implement the multi-label case.